lib/util.js

```diff
@@ -461,9 +463,16 @@ function formatValue(ctx, value, recurseTimes, ln) {
   }
 
   const keyLength = keys.length + symbols.length;
-  const constructor = getConstructorOf(value);
-  const ctorName = constructor && constructor.name ?
-    `${constructor.name} ` : '';
+
+  // try to use @@toStringTag, fall back to constructor name, empty string
+  let tag = '';
+  if (hasOwnProperty(value, Symbol.toStringTag)) {
```

Comment:
`@@toStringTag` is inherited. E.g. `Object.prototype.toString.call(new Date())` returns `[object Date]`, even though `new Date().hasOwnProperty(Symbol.toStringTag)` is false. IMO the check for `@@toStringTag` should be folded into a part of `getConstructorOf` and searched at the same time by walking the prototype chain (of course, then `getConstructorOf` would have to be modified to return a string instead).

Reply:
how about `getIdentifiableTagOf`

Reply:
Sure.